src/features/courses/course-controller.ts

```diff
@@ -875,20 +880,32 @@ class CourseController {
         } else {
             studentGrade = await workbook.getStudentGrade();
             if (studentGrade.courseWWTopicQuestionId !== options.questionId) {
-                throw new NotFoundError('The grade you have requested does not belong to the question provided');
+                throw new NotFoundError('The workbook you have requested does not belong to the question provided');
             }
         }
-        
 
-        if(_.isNil(workbook)) {
-            const workbooks = await studentGrade?.getWorkbooks({
-                limit: 1,
-                order: [ [ 'createdAt', 'DESC' ]]
-            });
-            workbook = workbooks?.[0] || null;
+        // if no workbookID, get the most recent workbook -- come back and delete this?
+        // if not enrolled, we don't even want to have workbooks
```

Comment:
If student grade is null (not enrolled) student grade would be null
since it is null it wouldn't actually call get workbooks, instead it would return undefined
then workbook would be null